### DISCUSSION TITLE
Disable only certain parts of text from being edited

### DISCUSSION THREAD

**Discussion by JaggerJo** [AvaloniaUI/AvaloniaEdit#397] [Q&A] [answered]
<img width="296" alt="Screenshot 2024-02-14 at 21 11 42" src="https://github.com/AvaloniaUI/AvaloniaEdit/assets/13090415/98421a7d-2398-4301-8042-bf67c2a480a1">

Text in the green rectangle should be editable. Text in the red box not. Let's assume it's as simple as checking if the line contains `=>` and if it does the text after it is read only (and styled differently). How would I prevent users from editing
certain parts of the text?

I've been implementing a custom `VisualLineElementGenerator` like explained [here](https://github.com/AvaloniaUI/AvaloniaEdit/discussions/366), so styling the text differently works.

**Comment by Gillibald** [comment, marked as answer]
https://github.com/AvaloniaUI/AvaloniaEdit/blob/cfaddb765a0d9318d34471fdc0005a2fa8dcc4af/src/AvaloniaEdit/Editing/TextArea.cs#L722

**Reply by JaggerJo** [reply]
Thanks!
